tests/test_diffraction_objects.py

```diff
@@ -232,13 +238,40 @@ def test_diffraction_objects_equality(inputs1, inputs2, expected):
     assert (diffraction_object1 == diffraction_object2) == expected
 
 
-def _test_valid_diffraction_objects(actual_diffraction_object, function, expected_array):
-    if actual_diffraction_object.wavelength is None:
-        with pytest.warns(UserWarning) as warn_record:
-            getattr(actual_diffraction_object, function)()
-        assert str(warn_record[0].message) == wavelength_warning_emsg
-    actual_array = getattr(actual_diffraction_object, function)()
-    return np.allclose(actual_array, expected_array)
+params_on_xtype = [
+    (
+        [
+            np.array([100, 200, 300, 400, 500, 600]),  # intensity array
+            np.array([0, 30, 60, 90, 120, 180]),  # tth array
+            np.array([1, 2, 3, 4, 5, 6]),  # q array
+            np.array([10, 20, 30, 40, 50, 60]),  # d array
+        ],
+        [
+            np.array([[0, 30, 60, 90, 120, 180], [100, 200, 300, 400, 500, 600]]),  # expected on_tth
+            np.array([[1, 2, 3, 4, 5, 6], [100, 200, 300, 400, 500, 600]]),  # expected on_q
+            np.array([[10, 20, 30, 40, 50, 60], [100, 200, 300, 400, 500, 600]]),  # expected on_d
+        ],
+    )
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params_on_xtype)
+def test_on_xtype(inputs, expected):
+    test = DiffractionObject()
+    test.on_tth = np.array([inputs[1], inputs[0]])
```

Comment:
This is not really how DO's should be used.  The tth, q and d arrays are somehow all different versions of the same array so we shouldn't be able to set them independently like this.  If we can now this is not desirable behavior, so we probably don't want a test that does this.    We should probably create a `set` property operator that generates all the arrays whenever a single one is specified in this way.  But for sure, we don't want a test with this undesirable behavior....